compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -2198,7 +2198,17 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
     auto cond_graph_inputs = cond_graph->inputs();
     auto cond_graph_input = cond_graph_inputs->at(cond_in->index());
 
-    return loco::NodeShape{*cond_graph_input->shape()};
+    auto cond_graph_input_shape = *cond_graph_input->shape();
+    auto this_shape = own_shape(node);
+
+    if (!(this_shape == cond_graph_input_shape))
```

Comment:
Is there a case when this happens?

I thought this will not happen.

Reply:
Yes, out in-house model has different shape. where shape for Cond has `[]` but main `WHILE` has `[1,0]`

Reply:
Look https://github.com/Samsung/ONE/issues/2260#issuecomment-644474454